includes/class-wc-payments-account.php

```diff
@@ -116,22 +116,28 @@ public function is_stripe_connected( bool $on_error = false ): bool {
 	}
 
 	/**
-	 * Checks if the account has been rejected, assumes the value of $on_error on server error.
+	 * Checks if the account has been rejected, assumes the value of $on_error on any account retrieval error.
+	 * Returns false if the account is not connected.
 	 *
 	 * @param bool $on_error Value to return on server error, defaults to false.
 	 *
-	 * @return bool True if the account is rejected, false otherwise, $on_error on error.
+	 * @return bool True if the account is connected and rejected, false otherwise, $on_error on error.
 	 */
 	public function is_account_rejected( bool $on_error = false ): bool {
 		try {
 			$account = $this->get_cached_account_data();
 
-			if ( empty( $account ) ) {
-				// Empty means no account, so not rejected.
+			if ( false === $account ) {
```

Comment:
I believe the same logic is covered by `try_is_stripe_connected` below – should we just use `is_stripe_connected` for all of the conditions aside from the `rejected` check itself?

On another note, what do you think of moving this function so that those two are adjacent in the code as before?